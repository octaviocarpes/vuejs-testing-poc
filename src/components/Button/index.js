export default {
	data() {
		return {
			buttonColor: false
		};
	},
	methods: {
		changeButtonColor() {
			this.buttonColor = !this.buttonColor;
		}
	},
	computed: {
		buttonStyle() {
			return this.buttonColor;
		}
	}
};