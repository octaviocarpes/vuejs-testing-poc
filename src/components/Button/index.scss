.button-container {
  button {
    padding: 10px;
    color: white;
    font-size: 16px;
    text-align: center;
    font-weight: bold;
    border-radius: 5px;
    border: none;

    &:hover {
      cursor: pointer;
    }

    &:focus {
      outline: none;
    }
  }

  .red {
    background-color: #f63f3f;
  }

  .blue {
    background-color: #23a8fc;
  }
}