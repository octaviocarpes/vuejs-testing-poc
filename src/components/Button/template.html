<div class="button-container">
    <button
            v-on:click="changeButtonColor"
            :class="[buttonStyle ? 'blue' : 'red']"
    >
        I am a button
    </button>
</div>