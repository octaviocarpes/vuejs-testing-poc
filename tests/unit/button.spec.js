import { render, fireEvent } from '@testing-library/vue';
import { mount } from '@vue/test-utils';
import Button from '@/components/Button/Button';

describe('Button component tests', () => {
	test('Button snapshot test', () => {
		const wrapper = mount(Button);
		expect(wrapper).toMatchSnapshot();
	});

	test('Button internal state', () => {
		const wrapper = mount(Button);
		expect(wrapper.vm).toBeTruthy();
		expect(typeof Button.data).toBe('function');
		expect(wrapper.vm.buttonColor).toBe(false);
	});

	test('Button styling test', () => {
		const wrapper = mount(Button);
		expect(wrapper.classes()).toContain('button-container');
		expect(wrapper.text()).toContain('I am a button');
	});

	test('Button behavior test', async () => {
		const { getByText } = render(Button);
		const button = getByText('I am a button');

		await fireEvent.click(button);
		expect(button.classList).toContain('blue');
	});
});