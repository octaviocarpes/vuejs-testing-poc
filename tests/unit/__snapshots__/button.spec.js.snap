// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests Button snapshot test 1`] = `
<div class="button-container"><button class="red">
    I am a button
  </button></div>
`;
